### ISSUE TITLE
Get-AzureResource: Doesn't return resources based on tags

### ISSUE THREAD

**Issue by pcgeek86** [Azure/azure-powershell#230]
In the Azure Resource Manager module, version 0.8.14, when I use the `Get-AzureResource` command with only the `-Tag` parameter, no Azure resources are returned, even though I have resources tagged with the one I specify.

```
Switch-AzureMode -Name AzureResourceManager;
Select-AzureSubscription -SubscriptionName 'Visual Studio Ultimate with MSDN';
Get-AzureResource -Tag @{ name = 'Trevor'; };
```

As you can see, in the Azure Preview Portal, I created an Azure Website, and added a tag `name = Trevor`.

![image](https://cloud.githubusercontent.com/assets/466713/6515404/5b67f69a-c33e-11e4-81a7-aef15cd244c1.png)

Cheers,
Trevor Sullivan
Microsoft MVP: PowerShell
https://trevorsullivan.net
https://twitter.com/pcgeek86

**Comment by pcgeek86** [comment]
I just figured out that I was using the wrong syntax for this. Instead of the HashTable containing a single element, it needs to contain two elements: `Name` and `Value`. Could this command be simplified to use a single element in the HashTable insetad?

If that isn't an option, can we at least perform some validation to ensure that the HashTable matches the expected input? As of today, it silently fails when the proper HashTable keys are not present.

Cheers,
Trevor Sullivan
Microsoft MVP: PowerShell
http://trevorsullivan.net
http://twitter.com/pcgeek86

**Comment by markcowl** [comment]
@pcgeek86 Unfortunately, Tag is overloaded in the API - it is used for both tagging somethign with a single value (i..e 'Important') or two values (i.e. Priority=Important).  For this reason, it's represented as the API represents it.  It would not be a bad overload to provide some syntactic sugar over the API, however.  Assigning to the appropriate team to look at./

**Comment by pcgeek86** [comment]
@markcowl @vivsriaus Did this get solved already? It's been nearly a year, so I may have missed some changes to the module since then. 

Cheers,
Trevor Sullivan

**Comment by ravbhatnagar** [comment]
Similar to #726 , #397 . Moreover querying through tags in not supported in the Find-AzureRmResource* cmdlets. There is a related bug on that too.

**Comment by ravbhatnagar** [comment]
We have a PR ready for fixing the usability of tags. This will be addressed as part of that. Since it was a breaking change we have postponed it to June release and added a deprecation message to state about the coming changes.

**Comment by ravbhatnagar** [comment]
Since this turned out to be a change impacting a lot of cmdlets from different teams, we are pushing this out by a month. The reason is to fix this across the board for all Azure cmdlets and give some warning about the upcoming change to customers of all those different cmdlets. The warning we added in added the last release was only to a subset of these cmdlets. 

So in the June release, all the cmdlets using this param will show this warning and then we will actually change it in the July release.

**Comment by pcgeek86** [comment]
That's interesting. I wonder why the warning was only added to a subset of cmdlets, instead of all affected ones.

**Comment by ravbhatnagar** [comment]
@pcgeek86 - Well, you can say that it took a little longer to get everyone on board with this breaking change :). But also, we wanted to do such breaking changes in a major version upgrade for Azure PS and club as many of such changes together. This was another major reason to push it out.

**Comment by pcgeek86** [comment]
It's interesting how some engineering teams (eg. Storage) make breaking changes without any warning, but when external customers ask for breaking changes, there are many barriers to get through. Double standards? :smile: 

I understand the desire to clump changes together. It's not a big deal, just interesting that it wasn't handled the first time. Good job being a "gatekeeper" and ensuring standards prior to release! I appreciate the improved focus, and I think it's safe to say that I speak for many other silent customer voices out there! :smile:

**Comment by ravbhatnagar** [comment]
Yes, I hear you. We are driving towards a more uniform standard across all teams in this respect too. Hopefully we will get better at this soon.

**Comment by ravbhatnagar** [comment]
This will be supported in Find-AzureRmResource, but currently does not work. Adding to Sept milestone

**Comment by markcowl** [comment]
@ravbhatnagar Please close if this is fixed.  Otherwise, please assign to the appropriate milestone.

**Comment by markcowl** [comment]
@ravbhatnagar  Please close this issue if it is fixed, otherwise please contact the customer and schedule in the appropriate milestone.

**Comment by ravbhatnagar** [comment]
This is currently unblocked in Find-AzureRmResource by using the tagName and tagValue params. We will add a Tag parameter to make it aligned with the uniform tag object that we have for all cmdlets
